src/py42/clients/cases.py

```diff
@@ -0,0 +1,135 @@
+class CasesClient(object):
+    """A client to expose cases API.
+
+    `Rest documentation <https://default-cases.core-int.cloud.code42.com/swagger-ui.html#/Cases>`__
+    """
+
+    def __init__(self, cases_service, cases_file_event_service):
+        self._cases_service = cases_service
+        self._file_events = cases_file_event_service
+
+    @property
+    def file_events(self):
+        """A collection of methods for managing file events associated with a given case.
+
+        Returns :
+            :class:`py42.services.casesfileevents.FileEventsService`
+        """
+        return self._file_events
+
+    def create(
+        self, name, subject=None, assignee=None, description=None, findings=None
+    ):
+        """Creates a new case.
+
+        Args:
+            name (str): Name of the case.
+            subject (str, optional): User UID of a subject of a case.
+            assignee (str, optional): User UID of the assignee.
+            description (str, optional): Description of the case
+            findings (str, optional): Observations of the case.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        return self._cases_service.create(
+            name, subject, assignee, description, findings
+        )
+
+    def get_all(
+        self,
+        name=None,
+        status=None,
+        created_at=None,
+        updated_at=None,
```

Comment:
@unparalleled-js @alanag13  @timabrmsn 
`created_at` and `updated_at` are ranges separated by '\', wondering whether we should make it simpler for user by breaking it into 4 fields, `created_at_from` , `created_at_to`  and `updated_at_from`, `updated_at_to`.

The names are just suggestions, open to other naming options.
Also the `datetime` format expected by the API is ISO format, do we want to follow that ?

Reply:
the pattern that we have in the rest of the codebase would mean that these fields should be named `min_create_time`, `max_create_time`, etc, but I agree that this is a good idea.

We should allow the string representation of whatever the API expects to be used.

Reply:
Done.